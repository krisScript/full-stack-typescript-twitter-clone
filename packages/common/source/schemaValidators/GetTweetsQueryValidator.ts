import * as yup from 'yup';

interface GetTweetsQueryValidatorType {
  sort: 'top' | 'trending' | 'new' | 'comments';
  confirmPassword: string;
}

const GetTweetsQueryValidator = yup
  .object<GetTweetsQueryValidatorType>()
  .shape({
    sort: yup
      .string()
      .trim()
      .oneOf(['top', 'trending', 'new', 'comments']),
    limit: yup
      .number()
      .integer()
      .min(1)
      .max(50),
    page: yup
      .number()
      .min(1)
      .integer(),
  });

export default GetTweetsQueryValidator;
