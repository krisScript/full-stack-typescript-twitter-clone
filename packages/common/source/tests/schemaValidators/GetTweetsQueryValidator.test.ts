import GetTweetsQueryValidator from '@schemaValidators/GetTweetsQueryValidator';

describe('GetTweetsQueryValidator', (): void => {
  const limit = 25;
  const page = 1;
  it(`should validate successfully`, async (): Promise<void> => {
    const sort = 'top';
    const query = { page, limit, sort };
    await expect(GetTweetsQueryValidator.validate(query)).resolves.toBe(query);
  });
  it(`should validate successfully`, async (): Promise<void> => {
    const sort = 'new';
    const query = { page, limit, sort };
    await expect(GetTweetsQueryValidator.validate(query)).resolves.toBe(query);
  });
  it(`should validate successfully`, async (): Promise<void> => {
    const sort = 'comments';
    const query = { page, limit, sort };
    await expect(GetTweetsQueryValidator.validate(query)).resolves.toBe(query);
  });
  it(`should validate successfully`, async (): Promise<void> => {
    const sort = 'trending';
    const query = { page, limit, sort };
    await expect(GetTweetsQueryValidator.validate(query)).resolves.toBe(query);
  });
  it(`should throw an error`, async (): Promise<void> => {
    const sort = 'invalid';
    const query = { page: 0, limit: 0, sort };
    await expect(
      GetTweetsQueryValidator.validate(query),
    ).rejects.toMatchSnapshot();
  });
});
