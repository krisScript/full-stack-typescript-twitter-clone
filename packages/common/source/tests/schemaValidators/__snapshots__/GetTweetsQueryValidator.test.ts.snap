// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetTweetsQueryValidator should throw an error 1`] = `[ValidationError: sort must be one of the following values: top, trending, new, comments]`;
